File: academia-dotNet/desafios/crud-com-plain-text/desafio-instituicao-wf/models/pessoas/Pessoa.cs
namespace desafio_instituicao_wf.models.pessoas;

public abstract class Pessoa
{
    public string Nome { get; private set; }
    public string Telefone { get; private set; }
    public string Cidade { get; private set; }
    public string Rg { get; }
    public string Cpf { get; }

    public Pessoa(string nome, string telefone, string cidade, string rg, string cpf)
    {
        Nome = nome;
        Telefone = telefone;
        Cidade = cidade;
        Rg = rg;
        Cpf = cpf;
    }

    public override string ToString()
    {
        return $"Titular: {Nome,-15} | Telefone: {Telefone,-12} | Cidade: {Cidade,-20} | CPF: {Cpf,-13} | RG: {Rg,-13}\n";
    }
}

File: alura-cursos/ano-2024/java/java-api-httpclient/src/models/Pessoa.java
package models;

import mapping.PessoaMapping;

public class Pessoa {

    String nome;
    int idade;
    String cidade;
    String estado;

    public Pessoa(PessoaMapping pessoaMapping) {
        this.nome = pessoaMapping.nome();
        this.idade = pessoaMapping.idade();
        this.cidade = pessoaMapping.cidade();
        this.estado = pessoaMapping.uf();
    }

    @Override
    public String toString() {
        return "Pessoa [nome: " + nome + ", idade: " + idade + ", cidade: " + cidade + ", estado: " + estado + "]";
    }
}
